common/djangoapps/third_party_auth/api/tests/test_permissions.py

```diff
@@ -0,0 +1,46 @@
+"""
+Tests for the Third Party Auth permissions
+"""
+import unittest
+import ddt
+from mock import Mock
+
+from rest_framework.test import APITestCase
+from django.conf import settings
+from third_party_auth.api.permissions import ThirdPartyAuthApiPermission
+
+from third_party_auth.tests.testutil import ThirdPartyAuthTestMixin
+
+IDP_SLUG_TESTSHIB = 'testshib'
+PROVIDER_ID_TESTSHIB = 'saml-' + IDP_SLUG_TESTSHIB
+
+
+@ddt.ddt
+@unittest.skipUnless(settings.ROOT_URLCONF == 'lms.urls', 'Test only valid in lms')
+class ThirdPartyAuthApiPermissionTest(ThirdPartyAuthTestMixin, APITestCase):
+    """ Tests for third party auth API permission """
+    def setUp(self):
+        """ Create users and oauth client for use in the tests """
+        super(ThirdPartyAuthApiPermissionTest, self).setUp()
+
+        self.enable_saml()
+        self.configure_saml_provider(name='TestShib', enabled=True, idp_slug=IDP_SLUG_TESTSHIB)
+        client = self.configure_oauth_client()
+        self.configure_api_permission(client, PROVIDER_ID_TESTSHIB)
+
+    @ddt.data(
+        (1, PROVIDER_ID_TESTSHIB, True),
+        (1, 'invalid-provider-id', False),
+        (999, PROVIDER_ID_TESTSHIB, False),
+        (999, 'invalid-provider-id', False),
+        (1, None, False),
```

Comment:
Thanks for having negative test cases for invalid providers.
Can you also add a negative test case for an existing valid provider but that is not accessible by the requesting client?